src/containers/NonBitGoRecoveryCoin/DogecoinForm.tsx

```diff
@@ -0,0 +1,150 @@
+import { Form, FormikHelpers, FormikProvider, useFormik } from 'formik';
+import { Link } from 'react-router-dom';
+import * as Yup from 'yup';
+import {
+  Button,
+  FormikSelectfield,
+  FormikTextarea,
+  FormikTextfield,
+} from '~/components';
+
+const validationSchema = Yup.object({
+  krsProvider: Yup.string()
+    .oneOf(['keyternal', 'bitgoKRSv2', 'dai'])
+    .label('Key Recovery Service'),
+  apiKey: Yup.string().required(),
+  userKey: Yup.string().required(),
+  backupKey: Yup.string().required(),
+  bitgoKey: Yup.string().required(),
+  walletPassphrase: Yup.string().required(),
+  recoveryDestination: Yup.string().required(),
+  scan: Yup.number().required(),
+}).required();
+
+export type DogecoinFormProps = {
+  onSubmit: (
+    values: DogecoinFormValues,
+    formikHelpers: FormikHelpers<DogecoinFormValues>
+  ) => void | Promise<void>;
+};
+
+type DogecoinFormValues = Yup.Asserts<typeof validationSchema>;
+
+export function DogecoinForm({ onSubmit }: DogecoinFormProps) {
+  const formik = useFormik<DogecoinFormValues>({
+    onSubmit,
+    initialValues: {
+      apiKey: '',
+      userKey: '',
+      backupKey: '',
+      bitgoKey: '',
+      walletPassphrase: '',
+      recoveryDestination: '',
+      scan: 20,
+      krsProvider: '',
+    },
+    validationSchema,
+  });
+
+  const backupKeyHelperText =
+    formik.values.krsProvider === ''
+      ? 'Your encrypted backup key, as found on your recovery KeyCard.'
+      : 'The backup public key for the wallet, as found on your recovery KeyCard.';
+
+  return (
+    <FormikProvider value={formik}>
+      <Form>
+        <h4 className="tw-text-body tw-font-semibold tw-border-b-0.5 tw-border-solid tw-border-gray-700 tw-mb-4">
+          Self-managed hot wallet details
+        </h4>
+        {/* <div className="tw-mb-4">
+          <FormikSelectfield
+            HelperText="The Key Recovery Service that you chose to manage your backup key. If you have the encrypted backup key, you may leave this blank."
+            Label="Key Recovery Service"
+            name="krsProvider"
+            Width="fill"
+          >
+            <option value="">None</option>
+            <option value="keyternal">Keyternal</option>
+            <option value="bitgoKRSv2">BitGo KRS</option>
+            <option value="dai">Coincover</option>
```

Comment:
Support for these providers is not yet implemeted for Doge coin. I believe there is a plan to add support for these soon. Hence, added them inside a comment.

Reply:
I'm guessing you copy/pasted this from other forms, I would suggest leaving it out, until its needed and copy/paste it again then.